tests/other/metadce/libcxxabi_message.cpp

```diff
@@ -0,0 +1,16 @@
+class Foo {
+public:
+  Foo() : counter(5) {}
+  int counter;
+};
+
+int testfunc() {
+  // libcxxabi will emit some checks for proper static initialization, which
+  // can call abort_message()
+  static Foo foo;
+  return ++foo.counter;
+}
```

Comment:
Can you explain what this code is doing exactly to cause the abort?  Is there some way would exercise `abort_message` more directly?    (Or simply this code.. at the very least you don't testfunc to be outside of main right?)

Also, should we run this also and check for its output on the console?

Reply:
Oh, this code is not causing an abort, but just compiles in code that contains a reference to the abort. There are lots of ways in libcxxabi that lead to `abort()` or `abort_message()` being called.. Search for `ABORT_WITH_MESSAGE()` -> `abort_message()` -> `printf()`

Reply:
Can you elaborate a little in your comment here @kripken.

Something like: "The point of this test is simply to pull in enough to libc++abi to include its internal `abort_message` code.  We want to ensure that (in release builds) this does not transitively pull in printf formatting code." ?

Reply:
Sure, done now.